projects/gateway2/query/query.go

```diff
@@ -126,18 +127,45 @@ type GatewayQueries interface {
 
 	GetLocalObjRef(ctx context.Context, from From, localObjRef apiv1.LocalObjectReference) (client.Object, error)
 
-	// GetRoutesForGateway finds the top level xRoutes attached to the provided Gateway
-	GetRoutesForGateway(ctx context.Context, gw *apiv1.Gateway) (*RoutesForGwResult, error)
 	// GetRouteChain resolves backends and delegated routes for a the provided xRoute object
 	GetRouteChain(ctx context.Context, obj client.Object, hostnames []string, parentRef apiv1.ParentReference) *RouteInfo
+
+	// GetRoutesForConsolidatedGateway finds the top level xRoutes attached to the provided Gateway and associated ListenerSet
+	GetRoutesForConsolidatedGateway(ctx context.Context, cgw *translator_types.ConsolidatedGateway) (*RoutesForGwResult, error)
+
+	// ConsolidateGateway returns the consolidated gateway that includes the list of listener sets mapped to it
+	ConsolidateGateway(ctx context.Context, gateway *apiv1.Gateway) (*translator_types.ConsolidatedGateway, error)
 }
 
 type RoutesForGwResult struct {
-	// key is listener name
-	ListenerResults map[string]*ListenerResult
+	// key is <parent.Namespace/parent.Name/listener.Name>
+	listenerResults map[string]*ListenerResult
```

Comment:
Don't directly expose this to ensure the right key is always generated via the getter and setter